projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -2166,7 +2166,15 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
 
         const state = directive.state;
         const start = state.startIndex;
-        const size = state.chunkSize - 1;
+        const isColumn = directive.igxForScrollOrientation === 'horizontal';
+
+        let size;
+        if (isColumn) {
+            const firstDataRow = this.rowList.find(r => r.cells);
+            size = firstDataRow.cells.length - 2;
```

Comment:
Why `- 2`?

Reply:
@ChronosSF Please look into providing an API method that returns how many cells are fully in view, or returns a floating point for the cells fully in view - like `5.6` when 60% of the last cell is visible.

Reply:
@DDavidkov Instead of branching the logic couldn't we just always scroll to the matching item index? 
For example: 
    private scrollDirective(directive: IgxForOfDirective<any>, goal: number): void {
        if (!directive) {
            return;
        }
        directive.scrollTo(goal);
    }

Reply:
@MayaKirova Not really. Technically, we can, but if we did, and you need to scroll just one row/column and we just call directive.scrollTo(goal) it would scroll the whole grid, so that the goal is the first visible row/column. We should try to scroll as little as possible, which means that if we need to scroll downwards or to the right, the cell with the active highlight should end up on the last visible row/column.

Reply:
@DDavidkov Thanks for the clarification. I've done some changes that should achieve the behavior you have described. Please review them.